File: CarthageObjC/CarthageObjC/ViewController.m
//
//  ViewController.m
//  CarthageObjC
//
//  Created by Wang Wei on 2018/8/29.
//  Copyright © 2018年 LINE Corp. All rights reserved.
//

#import "ViewController.h"
@import LineSDKObjC;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

    NSSet *permission = [NSSet setWithObjects:
                         [LineSDKLoginPermission profile],
                         [LineSDKLoginPermission openID],
                         nil];
    [[LineSDKLoginManager sharedManager]
        loginWithPermissions:permission
            inViewController:self
                     options:nil
           completionHandler:^(LineSDKLoginResult *result, NSError *error) {
               if (result) {
                   NSLog(@"User Name: %@", result.userProfile.displayName);
               } else {
                   NSLog(@"Error: %@", error);
               }
           }
     ];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: BinaryFramework/BinaryFramework/ViewController.swift
//
//  ViewController.swift
//  BinaryFramework
//
//  Created by Wang Wei on 2019/07/24.
//  Copyright © 2019 OneV's Den. All rights reserved.
//

import UIKit
import LineSDK

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.

        print(Constant.SDKVersion)
        LoginManager.shared.setup(channelID: "123", universalLinkURL: nil)
        let permission: Set<LoginPermission> = [.profile, .openID]
        LoginManager.shared.login(permissions: permission) { r in
            print(r)
        }
    }
}

